File: app/src/main/java/ua/com/foxminded/locationtrackera/model/locations/LocationRepository.java
package ua.com.foxminded.locationtrackera.model.locations;

import java.util.List;

import io.reactivex.rxjava3.core.Single;
import ua.com.foxminded.locationtrackera.util.Result;

public interface LocationRepository {

    void saveLocation(UserLocation userLocation);

    void deleteLocationsFromDb();

    List<UserLocation> getAllLocations();

    Single<Result<Void>> sendLocations(List<UserLocation> locationList);

    Single<Result<List<UserLocation>>> retrieveLocations(double startDate, double endDate);
}

File: models/src/main/java/ua/com/foxminded/locationtrackera/models/locations/LocationRepository.kt
package ua.com.foxminded.locationtrackera.models.locations

import io.reactivex.rxjava3.core.Single
import ua.com.foxminded.locationtrackera.models.util.Result

interface LocationRepository {

    fun saveLocation(userLocation: UserLocation)

    fun deleteLocationsFromDb()

    fun getAllLocations(): List<UserLocation>

    fun sendLocations(locationList: List<UserLocation>): Single<Result<Void?>>

    fun retrieveLocations(fromTime: Long, toTime: Long): Single<Result<List<UserLocation>>>
}
